### ISSUE TITLE
Ability to add custom directories to scan for JREs

### ISSUE THREAD

**Issue by he3als** [PolyMC/PolyMC#668]
### Role

I use Scoop (a Windows package manager) to install and update Java. I use multiple versions of Java with PolyMC.

### Suggestion

I want PolyMC to have a section in the Java settings to add custom paths to scan for JREs.

### Benefit

So that I do not have to manually type out or find my JREs.

### This suggestion is unique

- [X] I have searched the issue tracker and did not find an issue describing my suggestion, especially not one that has been rejected.

### You may use the editor below to elaborate further.

I made an okay-ish concept a while ago of what I was thinking about it looking like. Scanning recursively would be good, but scanning only the folder selected for a JRE is fine too.

![image](https://user-images.githubusercontent.com/65787561/170860914-de427de0-bed4-4663-ad36-324679c4a81c.png)

**Comment by Technous285** [comment]
As a PolyMC user with Java 8 & 17 installed; I've found using the Auto-detect function (shown there in the image) automatically scans for any JRE's you have installed and shows them as a list for you to pick one as a default.
Said default JRE can be overruled on a instance-by-instance manner using the same Auto-detect function from within the Settings section of each instance's "Edit Instance" feature, by a simple checkbox next to the line "Java Installation" on that section's screen.

<NB: Any instance section shown minimized was imported from a previous MultiMC install, all made prior to the current shenanigans with CF.>
![PolyMC Launcher Java Settings](https://user-images.githubusercontent.com/15863/170877231-6867e370-f031-4e7e-8f06-6e3a3c7bad49.png)
![PolyMC Instance Java Settings](https://user-images.githubusercontent.com/15863/170877227-71afe40d-5087-4724-b9b0-596a0a6cfd8a.png)

**Comment by he3als** [comment]
Yes, but that is because PolyMC scans certain registry keys for different Java versions. However, you can't add custom JREs to that list in the auto detect menu.

**Comment by Technous285** [comment]
"Custom" as in "not published by Oracle, but rather by a community project" y'mean, aside from OpenJDK?
That's a kinda different kettle of fish to just being able to direct to JRE's not in the standard `C:/Program Files/` location that your screenshot seemed to imply, imo.

Either way; I would agree to having an option (within that Java version selection window at least) to scan for JRE's aside from Oracle and OpenJDK by pointing to a folder outside of `C:/Program Files/` if they're not located there (at least if they return the calls of being compatible with Oracle and OpenJDK JRE's, though I will admit I haven't messed around with JRE's other than Oracle's and OpenJDK).

**Comment by Scrumplex** [comment]
Technically this is already a feature, just hidden. You can set the `POLYMC_JAVA_PATHS` environment variable to a list of predefined java executables to scan. 
On Windows you would set this as 
`POLYMC_JAVA_PATHS=C:\path\to\java18\java.exe;C:\path\to\java8\java.exe`

On other platforms, the delimiter would be `:` instead of `;`

**Comment by Technous285** [comment]
> Technically this is already a feature, just hidden. You can set the `POLYMC_JAVA_PATHS` environment variable to a list of predefined java executables to scan. On Windows you would set this as `POLYMC_JAVA_PATHS=C:\path\to\java18\java.exe;C:\path\to\java8\java.exe`
> 
> On other platforms, the delimiter would be `:` instead of `;`

Awesome that it's technically there even if hidden, would it be possible to implement that within the GUI for a more user-friendly experience for those not comfortable delving in the files and setting things there?

**Comment by he3als** [comment]
Yeah, this suggestion was asking for that hidden option to have something in the GUI. What I mean by non-standard Java installations are Java installs that do not add registry keys that PolyMC scans for. A Windows package manager called [Scoop](https://www.scoop.sh) doesn't add these registry keys when installing Java through it, as it installs portable versions of applications without admin privledges. Also, I do have some other installations of Java (GraalVM) that PolyMC would never look for.

Also, if you set that environment variable to a path like Program Files, would it recursively search folders? As in, would it look at every single folder inside of Program Files for a Java install or just in the root of Program Files?

**Comment by he3als** [comment]
> Technically this is already a feature, just hidden. You can set the `POLYMC_JAVA_PATHS` environment variable to a list of predefined java executables to scan. 
> 
> On Windows you would set this as 
> 
> `POLYMC_JAVA_PATHS=C:\path\to\java18\java.exe;C:\path\to\java8\java.exe`
> 
> 
> 
> On other platforms, the delimiter would be `:` instead of `;`

I tried setting a user environment variable for this, but it never worked.
